sdk/core/core/src/az_span.c

```diff
@@ -618,3 +618,88 @@ _az_span_scan_until(az_span self, _az_predicate predicate, int32_t* out_index)
   }
   return AZ_ERROR_ITEM_NOT_FOUND;
 }
+
+AZ_NODISCARD az_span _az_span_trim_white_space(az_span source)
+{
+  // az_span source with size 0 is allowed. For instance, a header value can be AZ_SPAN_NULL
+  if (az_span_size(source) == 0)
+  {
+    return source;
+  }
+
+  // remove white spaces from left
+  az_span left_trim = _az_span_trim_white_space_from_start(source);
+  // calculate the offset after moving after white spaces
+  int32_t offset = az_span_size(source) - az_span_size(left_trim);
+  // remove from right
+  az_span right_trim = _az_span_trim_white_space_from_end(source);
+
+  if (offset == 0)
+  {
+    // nothing removed from left, return
+    return left_trim;
```

Comment:
I don't think this is correct. It is missing the `right_trim`, even if nothing was removed from left. Please add unit tests to verify correctness.

Reply:
it should be `return right_trim;`  it's a typo